test/Microsoft.AspNetCore.Mvc.FunctionalTests/TestingInfrastructureInheritanceTests.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) .NET  Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.AspNetCore.TestHost;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.FunctionalTests
+{
+    public class TestingInfrastructureInheritanceTests
+    {
+        [Fact]
+        public void TestingInfrastructure_WithWebHostBuilderRespectsCustomizations()
+        {
+            // Act
+            var factory = new CustomizedFactory<BasicWebSite.Startup>();
+            var customized = factory
+                .WithWebHostBuilder(builder => factory.ConfigureWebHostCalled.Add("Customization"))
+                .WithWebHostBuilder(builder => factory.ConfigureWebHostCalled.Add("FurtherCustomization"));
```

Comment:
This doesn't look realistic. When would a test class call `WithWebHostBuilder(...)` twice?

Reply:
Once on the constructor for the class another one on the test to further customize a specific scenario.

Reply:
:ok: